great_expectations/experimental/datasources/sqlite_datasource.py

```diff
@@ -0,0 +1,121 @@
+from __future__ import annotations
+
+from datetime import datetime
+from typing import TYPE_CHECKING, Dict, List, Optional, Type
+
+import pydantic
+from pydantic import dataclasses as pydantic_dc
+from typing_extensions import ClassVar
+
+from great_expectations.experimental.datasources.interfaces import DataAsset
+
+if TYPE_CHECKING:
+    import sqlalchemy
+
+from typing_extensions import Literal
+
+from great_expectations.experimental.datasources.sql_datasource import (
+    BatchSortersDefinition,
+    ColumnSplitter,
+    DatetimeRange,
+    SQLDatasource,
+    SQLDatasourceError,
+    TableAsset,
+    _query_for_year_and_month,
+)
+
+
+class SqliteTableAsset(TableAsset):
+    # Subclass overrides
+    type: Literal["sqlite_table"] = "sqlite_table"  # type: ignore[assignment]
+    column_splitter: Optional[SqliteYearMonthSplitter] = None  # type: ignore[assignment]
+
+    # This asset type will support a variety of splitters
+    def add_year_and_month_splitter(
+        self,
+        column_name: str,
+    ) -> SqliteTableAsset:
+        """Associates a year month splitter with this sqlite table data asset
+
+        Args:
+            column_name: A column name of the date column where year and month will be parsed out.
+
+        Returns:
+            This SqliteTableAsset so we can use this method fluently.
+        """
+        self.column_splitter = SqliteYearMonthSplitter(
+            column_name=column_name,
+        )
+        return self
+
+
+@pydantic_dc.dataclass(frozen=True)
+class SqliteYearMonthSplitter(ColumnSplitter):
+    method_name: Literal["split_on_year_and_month"] = "split_on_year_and_month"
+    param_names: List[Literal["year", "month"]] = pydantic.Field(
+        default_factory=lambda: ["year", "month"]
+    )
+
+    def param_defaults(self, table_asset: TableAsset) -> Dict[str, List]:
```

Comment:
I tried making the `table_asset` input parameter more specific (ie `SqliteTableAsset`) using `Generics` but ran into some problems using pydantic and `Generics` together. I decided not to resolve the issue right now.